pkg/util/pki/generate.go

```diff
@@ -72,6 +76,14 @@ func GeneratePrivateKeyForCertificate(crt *v1.Certificate) (crypto.Signer, error
 		}
 
 		return GenerateECPrivateKey(keySize)
+	case v1.Ed25519KeyAlgorithm:
+		keySize := Ed25519
+
+		if crt.Spec.PrivateKey.Size > 0 {
+			keySize = crt.Spec.PrivateKey.Size
+		}
+
+		return GenerateEdPrivateKey(keySize)
```

Comment:
suggestion: as with `GenerateEdPrivateKey` below, I think anything involving keysize here is redundant:
```suggestion
	case v1.Ed25519KeyAlgorithm:
		return GenerateEd25519PrivateKey()
```